app/controllers/chargeback_assignment_controller.rb

```diff
@@ -4,110 +4,83 @@ class ChargebackAssignmentController < ApplicationController
       changed = (@edit[:new].except(*except) != @edit[:current].except(*except))
-      page.replace("cb_assignment_div", :partial => "cb_assignments") if params[:cbshow_typ] || params[:cbtag_cat] || params[:cblabel_key]
+      prefix = @edit[:new][:type].downcase
+      page.replace("#{prefix}_assignments_div", :partial => "#{prefix}_assignments") if params[:cbshow_typ] || params[:cbtag_cat] || params[:cblabel_key]
       page << javascript_for_miq_button_visibility(changed)
     end
   end
 
-  def cb_assign_update
+  def update
+    clear_flash_msg
+    return unless load_edit("cbassign_edit__#{params[:id]}", "index")
     if params[:button] == "reset"
-      get_node_info(x_node)
-      add_flash(_("All changes have been reset"), :warning)
-      replace_right_cell
-    else
-      return unless load_edit("cbassign_edit__#{x_node}", "replace_cell__chargeback")
-
-      cb_assign_set_record_vars
-      rate_type = x_node.split('-').last
+      set_form_vars
+      flash_to_session(_("All changes have been reset"), :warning)
+    elsif params[:button] == "save"
+      set_record_vars
+      rate_type = params[:id]
       begin
         ChargebackRate.set_assignments(rate_type, @edit[:set_assignments])
       rescue StandardError => bang
         render_flash(_("Error during 'Rate assignments': %{error_message}") % {:error_message => bang.message}, :error)
       else
-        add_flash(_("Rate Assignments saved"))
-        get_node_info(x_node)
-        replace_right_cell
+        flash_to_session(_("Rate Assignments saved"))
       end
+    else
+      flash_to_session("Rate Assignment has been cancelled")
+    end
+    set_form_vars unless params[:button] == 'reset'
+    render :update do |page|
+      page << javascript_prologue
+      prefix = @edit[:new][:type].downcase
+      page.replace("#{prefix}_assignments_div", :partial => "#{prefix}_assignments")
     end
-  end
-
-  def title
-    @title = _("Chargeback Assignment")
   end
 
   private ############################
 
-  def features
-    [
-      {
-        :role  => "chargeback_assignments",
-        :name  => :cb_assignments,
-        :title => _("Assignments")
-      }
-    ].map { |hsh| ApplicationController::Feature.new_with_hash(hsh) }
-  end
-
-  def get_node_info(node, show_list = true)
-    @show_list = show_list
-    node = valid_active_node(node)
-    if ["xx-Compute", "xx-Storage"].include?(node)
-      cb_assign_set_form_vars
-      @right_cell_text = case node
-                         when "xx-Compute" then _("Compute Rate Assignments")
-                         when "xx-Storage" then _("Storage Rate Assignments")
-                         end
-    else
-      @right_cell_text = _("All Assignments")
-    end
-    {:view => @view, :pages => @pages}
+  def build_tabs
+    @breadcrumbs = []
+    @active_tab = @tabform
+    @tabs = []
+    @tabs.push(["Compute", _("Compute")])
+    @tabs.push(["Storage", _("Storage")])
```

Comment:
Do we need to do it like this? Can't this be just a simple static array without pushing data into it afterwards?